doc/source/guide/derivatives_guide.rst

```diff
@@ -0,0 +1,151 @@
+
+Rules for the Fréchet derivative
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Many of the usual rules for derivatives also hold for the Fréchet derivative, i.e.
+
+* Linearity
+
+.. math::
+    (a f + b y)'(x)(y) = a f'(x)(y) + b g'(x)(y)
+
+* Chain rule
+
+.. math::
+    (g \circ f)'(x)(y) = g'(f(x))(f'(x)(y))
+
+* Linear operators are their own derivatives. Assume :math:`f` linear, then:
+
+.. math::
+   f'(x)(y) = f(y)
+
+Implementations in ODL
+~~~~~~~~~~~~~~~~~~~~~~
+
+* The derivative is implemented in ODL  for `Operator`'s via the `Operator.derivative` method.
+* It can be numerically computed using the `NumericalDerivative` operator.
+* Many of the operator arithmetic classes implement the usual rules for the Fréchet derivative, such as the chain rule, distributivity over addition etc.
+
+Gradient
+########
+In the classical setting of functionals :math:`f : \mathbb{R}^n \rightarrow \mathbb{R}`, the gradient is the vector
+
+.. math::
+    \nabla f =
+    \begin{bmatrix}
+        \dfrac{\partial f}{\partial x_1}
+        , \dots,
+        \dfrac{\partial f}{\partial x_n}
+    \end{bmatrix}
+
+This can be generalized to the setting of functionals :math:`f : \mathcal{X} \rightarrow \mathbb{R}` mapping elements in some Hilbert space :math:`\mathcal{X}` to the real numbers by noting that the derivative has a special form by the `Riesz representation theorem
+<https://en.wikipedia.org/wiki/Riesz_representation_theorem>`_, namely that the linear operator :math:`f'(x)` can be represented by a vector :math:`[\nabla f](x) \in \mathcal{X}` such that
+
+.. math::
+    f'(x)(y) = \langle y, [\nabla f](x) \rangle
+
+We call the (possibly nonlinear) operator :math:`x \rightarrow [\nabla f](x)` the *Gradient operator* of :math:`f`.
+
+Implementations in ODL
+~~~~~~~~~~~~~~~~~~~~~~
+
+* The gradient is implemented in ODL `Functional`'s via the `Functional.gradient` method.
+* It can be numerically computed using the `NumericalGradient` operator.
+
+Hessian
+#######
+In the classical setting of functionals :math:`f : \mathbb{R}^n \rightarrow \mathbb{R}`, the hessian in a point :math:`x` is the matrix :math:`H` such that
+
+.. math::
+    H(x) =
+    \begin{bmatrix}
+    \dfrac{\partial^2 f}{\partial x_1^2} & \dfrac{\partial^2 f}{\partial x_1\,\partial x_2} & \cdots & \dfrac{\partial^2 f}{\partial x_1\,\partial x_n} \\
+    \dfrac{\partial^2 f}{\partial x_2\,\partial x_1} & \dfrac{\partial^2 f}{\partial x_2^2} & \cdots & \dfrac{\partial^2 f}{\partial x_2\,\partial x_n} \\
+    \vdots & \vdots & \ddots & \vdots \\
+    \dfrac{\partial^2 f}{\partial x_n\,\partial x_1} & \dfrac{\partial^2 f}{\partial x_n\,\partial x_2} & \cdots & \dfrac{\partial^2 f}{\partial x_n^2}
+    \end{bmatrix}
+
+Where the derivatives are evaluated in the point :math:`x`. It has the property that that the quadratic variation of :math:`f` is
+
+.. math::
+    f(x + \Delta x) \approx f(x) + \langle \Delta x, [\nabla f](x)\rangle + \langle \Delta x, [H(x)](\Delta x)\rangle
```

Comment:
Why change notation to deltas? And why write approximate equality? I would clearly prefer using the same `d` as above and using O notation for higher order terms, making it an exact equality.